lib/matplotlib/rcsetup.py

```diff
@@ -438,19 +439,6 @@ def validate_ps_distiller(s):
         return ValidateInStrings('ps.usedistiller', ['ghostscript', 'xpdf'])(s)
 
 
-def _validate_papersize(s):
-    # Re-inline this validator when the 'auto' deprecation expires.
-    s = ValidateInStrings("ps.papersize",
-                          ["figure", "auto", "letter", "legal", "ledger",
-                           *[f"{ab}{i}" for ab in "ab" for i in range(11)]],
-                          ignorecase=True)(s)
-    if s == "auto":
-        _api.warn_deprecated("3.8", name="ps.papersize='auto'",
-                             addendum="Pass an explicit paper type, figure, or omit "
-                             "the *ps.papersize* rcParam entirely.")
-    return s
-
-
```

Comment:
This removal seems unrelated to the hatch changes. And the replacement does not provide the warning that we do want to have there.